tests/unittests/test_exception_handling.py

```diff
@@ -173,9 +173,19 @@ def test_409_error(self, mocked_parse_args, mocked_request, mock_verify_access):
         # Verifying the message formed for the custom exception
         self.assertEqual(str(e.exception), "HTTP-error-code: 409, Error: The request could not be completed due to a conflict with the current state of the server.")
 
+    def test_501_error(self, mocked_parse_args, mocked_request, mock_verify_access):
+        mocked_request.return_value = get_response(501, raise_error = True)
+        test_client = GithubClient(self.config)
+
+        with self.assertRaises(tap_github.client.Server5xxError) as e:
+            test_client.authed_get("", "")
+
+        # Verifying the message formed for the custom exception
+        self.assertEqual(str(e.exception), "HTTP-error-code: 501, Error: Unknown Error")
```

Comment:
Add assertion of retry.

Reply:
Was having an assertion in timeout unittest, however combined in the same and removed that one